### PR TITLE
Added nodejs and npm for chrome extension

### LINKED ISSUE #161: Add node and npm to `.sail/Dockerfile`
This is because of the extension

### PR DESCRIPTION
**Pull request by roberthmiller**
This PR is to solve the issue when trying to develop for the chrome extension through sail.

Currently, when running `sail run roberthmiller/sail`, I am unable to use npm (getting the deps for the chrome extension). This PR is to fix that.

Edit:
Resolves #161

### CODE REVIEW

**Comment by roberthmiller** [comment]
Note: This currently fails. I am working on a fix for this.

**Comment by roberthmiller** [comment]
Ok, This is ready for review.

File: .sail/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM codercom/ubuntu-dev-go:latest
 
+RUN sudo apt-get update
```
Comment (nathanpotter) [inline .sail/Dockerfile]: Any `apt-get update` should be combined with an `apt-get install` in a single `RUN` operation.

See [dockerfile best practices - RUN](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)

File: .sail/Dockerfile [hunk 2]
```diff
@@ -8,7 +9,13 @@ ENV GO111MODULE=off
...
 # Install the latest version of Hugo.
 RUN wget -O /tmp/hugo.deb https://github.com/gohugoio/hugo/releases/download/v0.55.4/hugo_extended_0.55.4_Linux-64bit.deb && \
-    sudo dpkg -i /tmp/hugo.deb && \
-    rm -f /tmp/hugo.deb
+  sudo dpkg -i /tmp/hugo.deb && \
+  rm -f /tmp/hugo.deb
 
 RUN installext peterjausovec.vscode-docker
+
+RUN sudo apt-get install curl -y
```
Comment (nathanpotter) [inline .sail/Dockerfile]: curl is already installed in the `base` image

File: .sail/Dockerfile [hunk 3]
```diff
@@ -8,7 +9,13 @@ ENV GO111MODULE=off
...
+
+RUN sudo apt-get install software-properties-common -y && \
```
Comment (nathanpotter) [inline .sail/Dockerfile]: I think we should use `nvm` for installing node, see instructions [here](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-18-04#installing-using-nvm).

@kylecarbs what do you think?

**Comment by roberthmiller** [comment]
I will make some updates and you can squash and merge :smile:

**Comment by roberthmiller** [comment]
If it is decided to change this to NVM, can we do this in a later PR?

**Comment by nathanpotter** [comment]
I'd rather just use NVM in this PR and change it later if needed, as I'm pretty sure that's how Kyle will want it installed. Also, that'll remove the need to install `software-properties-common`.

**Comment by roberthmiller** [comment]
Ok, will update that after I am done submitting this other PR.

**Comment by roberthmiller** [comment]
@nathanpotter This PR should be good to go.

File: .sail/Dockerfile [hunk 4]
```diff
@@ -1,14 +1,17 @@
 FROM codercom/ubuntu-dev-go:latest
```
Comment (nathanpotter) [inline .sail/Dockerfile]: Change the shell to bash so we can source the `nvm.sh` later.
```suggestion
SHELL ["/bin/bash", "-c"]
```

File: .sail/Dockerfile [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
 
-RUN sudo apt-get install -y htop
+RUN sudo apt-get update && \
+  sudo apt-get install -y htop && \
+  curl -sL https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | sudo -E bash -
```
Comment (nathanpotter) [inline .sail/Dockerfile]: We should Install the latest nvm and install the default node/npm as well. We should also install without `sudo` and in a separate `RUN` to utilize docker build caching
```suggestion
RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash && \
    . ~/.nvm/nvm.sh \
    && nvm install node
```

Comment (roberthmiller) [reply .sail/Dockerfile]: didn't even see that there is a newer version, this was just from the link that you sent before.

**Comment by roberthmiller** [comment]
Good to go

File: .sail/Dockerfile [hunk 6]
```diff
@@ -1,14 +1,19 @@
 FROM codercom/ubuntu-dev-go:latest
+SHELL ["/bin/bash", "-c"]
+RUN sudo apt-get update && \
+  sudo apt-get install -y htop && \
```
Comment (nathanpotter) [inline .sail/Dockerfile]: Trailing `&& \`, will squash and merge afterwards
```suggestion
  sudo apt-get install -y htop
```

**Comment by roberthmiller** [comment]
:+1:

**Pull request merged**